crates/accelerate/src/two_qubit_decompose.rs

```diff
@@ -2760,19 +2757,9 @@ impl TwoQubitControlledUDecomposer {
         gates1.global_phase = global_phase;
         Ok(gates1)
     }
-}
 
-#[pymethods]
-impl TwoQubitControlledUDecomposer {
-    ///  Initialize the KAK decomposition.
-    ///  Args:
-    ///      rxx_equivalent_gate: Gate that is locally equivalent to an :class:`.RXXGate`:
-    ///      :math:`U \sim U_d(\alpha, 0, 0) \sim \text{Ctrl-U}` gate.
-    ///  Raises:
-    ///      QiskitError: If the gate is not locally equivalent to an :class:`.RXXGate`.
-    #[new]
-    #[pyo3(signature=(rxx_equivalent_gate))]
-    pub fn new(py: Python, rxx_equivalent_gate: RXXEquivalent) -> PyResult<Self> {
+    /// Initialize the KAK decomposition.
+    pub fn new_inner(rxx_equivalent_gate: RXXEquivalent, euler_basis: &str) -> PyResult<Self> {
```

Comment:
If you're using `new_inner` here is there a reason to just take the `EulerBasis` object directly?

Reply:
it's similar to ‎`TwoQubitBasisDecomposer::new_inner` which also gets `euler_basis: &str`
https://github.com/Qiskit/qiskit/blob/75436a485bfe25e8732fbb9a6cc8305275ecbfba/crates/accelerate/src/two_qubit_decompose.rs#L1724